Library/Homebrew/utils/analytics.sh

```diff
@@ -1,9 +1,26 @@
 setup-analytics() {
-  [[ -n "$HOMEBREW_NO_ANALYTICS" ]] && return
-
   # User UUID file. Used for Homebrew user counting. Can be deleted and
   # recreated with no adverse effect (beyond our user counts being inflated).
   HOMEBREW_ANALYTICS_USER_UUID_FILE="$HOME/.homebrew_analytics_user_uuid"
+
+  if [[ -n "$HOMEBREW_NO_ANALYTICS" ]]
+  then
+    rm -rf "$HOMEBREW_ANALYTICS_USER_UUID_FILE"
```

Comment:
I'd probably make this `rm -f` instead of `rm -rf`, just as a bit of safety measure in case there's a bug in execution somehow. `rm -rf` is a bit punchy if it borks.